workqueue/workqueue.go

```diff
@@ -267,6 +268,52 @@ func handleUpvote(ctx context.Context, db *sql.DB, attributes []sdk.Attribute, h
 	return tx.Commit()
 
 }
+
+// This works for both stake and unstake commands since they both send only the total amount, which is upserted.
+func handleStake(ctx context.Context, db *sql.DB, attributes []sdk.Attribute, height int64, ts time.Time) error {
+	attrs := parseAttributes(attributes)
+	vendorID, err := strconv.Atoi(attrs["vendor_id"])
+	if err != nil {
+		return err
+	}
+	amount, err := strconv.ParseInt(attrs["amount"], 10, 64)
+	if err != nil {
+		return err
+	}
+	postID := attrs["post_id"]
+	delegator := attrs["delegator"]
+	validator := attrs["validator"]
+
+	if amount == 0 {
+		s, err := models.Stakes(
+			models.StakeWhere.VendorID.EQ(vendorID),
+			models.StakeWhere.PostID.EQ(postID),
+			models.StakeWhere.Delegator.EQ(delegator),
+		).One(ctx, db)
+		if err != nil {
+			return err
+		}
+
+		rowAff, err := s.Delete(ctx, db)
+		if (rowAff == 0) || (err != nil) {
+			return err
+		}
+
+		return nil
+	}
+
+	model := &models.Stake{
+		Height:    height,
+		VendorID:  vendorID,
+		PostID:    postID,
+		Delegator: delegator,
+		Validator: validator,
+		Amount:    amount,
+	}
+
+	return model.Upsert(ctx, db, true, []string{"vendor_id", "post_id"}, boil.Whitelist("amount"), boil.Infer())
```

Comment:
Regarding your question about updated_at, you will need to add it to the whitelist `boil.Whitelist("amount", "updated_at")`